python/ray/util/client/__init__.py

```diff
@@ -67,6 +69,7 @@ def connect(self,
                 metadata=metadata,
                 connection_retries=connection_retries)
             self.api.worker = self.client_worker
+            self.client_worker._server_init(job_config)
```

Comment:
```suggestion
            if job_config is not None:
                self.client_worker._server_init(job_config)
```

Please add this change so that the change is backwards compatible for clients that do not specify job config. In that case we should ignore it entirely.

Reply:
Hmmm, actually, not all client has job_config. I'm thinking this case:

- server has job_config run
- client doesn't have this

We should prevent it from running I guess?

Reply:
We always need to call server_init to set it up. But I've made it backward compatible in server side.